cstar/base/component.py

```diff
@@ -158,3 +138,66 @@ def post_run(self) -> None:
         This abstract method will be implemented differently by different Component types.
         """
         pass
+
+
+class Discretization:
+    """
+    Holds discretization information about a Component.
+
+    Attributes:
+    -----------
+
+    time_step: int
+        The time step with which to run ROMS in this configuration
+    nx,ny,n_levels: int
+        The number of x and y points and vertical levels in the domain associated with this object
```

Comment:
It might break docs formatting to not list these properly

```suggestion
    nx: int
        The number of x points in the domain
    ny: int
        The number of y points in the domain
    n_levels: int
        The number of vertical levels in the domain
```

Reply:
Addressed in  b69800f70ca3a583ffdde906d663d847df97404b